File: 2913-subarrays-distinct-element-sum-of-squares-i/2913-subarrays-distinct-element-sum-of-squares-i.cpp
class Solution {
public:
    int sumCounts(vector<int>& nums) {
        int ans = 0;
        int n = nums.size();
        for(int i=0;i<n;i++){
            unordered_set<int> st;
            int distinct  = 0;
            for(int j=i;j<n;j++){
                if(st.find(nums[j])==st.end()){
                    distinct++;
                }
                st.insert(nums[j]);
                ans += (distinct *distinct);
            }
        }
        return ans;
    }
};

File: 2913-subarrays-distinct-element-sum-of-squares-i/2913-subarrays-distinct-element-sum-of-squares-i.py
class Solution:
    def sumCounts(self, nums: List[int]) -> int:
        curSum = 0

        for i in range(len(nums)):
            for j in range(i,len(nums) + 1):
                curSum += len(set(nums[i:j])) ** 2
        return curSum